spec/support/city_macros.rb

```diff
@@ -0,0 +1,30 @@
+module CityMacros
+
+  def create_city(city = build(:city))
+    navigate_cities_index
+
+    click_link 'New City'
+    fill_in 'Name', with: city.name
+    click_button 'Create City'
+  end
+
+  def update_city(city = build(:city))
+    navigate_cities_index
+    click_link('Edit', match: :first)
+    fill_in 'Name', with: city.name
+    click_button 'Update City'
+  end
+
+  def destroy_city
```

Comment:
same as before, better to have this explicit on the scenario, not worth to be extracted